db/migrate/20131029215548_add_spree_fields_to_custom_user_table.rb

```diff
@@ -1,7 +1,9 @@
 class AddSpreeFieldsToCustomUserTable < ActiveRecord::Migration
   def up
-    add_column "alchemy_users", :spree_api_key, :string, :limit => 48
-    add_column "alchemy_users", :ship_address_id, :integer
-    add_column "alchemy_users", :bill_address_id, :integer
+    if ::Spree.user_class == "Alchemy::User"
+      add_column "alchemy_users", :spree_api_key, :string, :limit => 48
+      add_column "alchemy_users", :ship_address_id, :integer
+      add_column "alchemy_users", :bill_address_id, :integer
+    end
   end
 end
```

Comment:
Since this migration is only needed for the `Alchemy::User` option, it should be generated only then.